python/semantic_kernel/connectors/memory/postgres/postgres_memory_store.py

```diff
@@ -0,0 +1,524 @@
+MAX_DIMENSIONALITY = 2000
+DEFAULT_SCHEMA = "public"
+
+
+class PostgresMemoryStore(MemoryStoreBase):
+    """A memory store that uses Postgres with pgvector as the backend."""
+
+    _connection_string: str
+    _connection_pool: ConnectionPool
+    _default_dimensionality: int
+    _schema: str
+    _logger: Logger
+
+    def __init__(
+        self,
+        connection_string: str,
+        default_dimensionality: int,
+        min_pool: int,
+        max_pool: int,
+        schema: str = DEFAULT_SCHEMA,
+        logger: Optional[Logger] = None,
+    ) -> None:
+        """Initializes a new instance of the PostgresMemoryStore class.
+
+        Arguments:
+            connection_string {str} -- The connection string to the Postgres database.
+            default_dimensionality {int} -- The default dimensionality of the embeddings.
+            min_pool {int} -- The minimum number of connections in the connection pool.
+            max_pool {int} -- The maximum number of connections in the connection pool.
+            schema {str} -- The schema to use. (default: {"public"})
+            logger {Optional[Logger]} -- The logger to use. (default: {None})
+        """
+        if default_dimensionality > MAX_DIMENSIONALITY:
+            raise ValueError(
+                f"Dimensionality of {default_dimensionality} exceeds "
+                + f"the maximum allowed value of {MAX_DIMENSIONALITY}."
+            )
+        if default_dimensionality <= 0:
+            raise ValueError("Dimensionality must be a positive integer. ")
+
+        self._connection_string = connection_string
+        self._default_dimensionality = default_dimensionality
+        self._connection_pool = ConnectionPool(
+            self._connection_string, min_size=min_pool, max_size=max_pool
+        )
+        self._schema = schema
+        atexit.register(self._connection_pool.close)
+        self._logger = logger or NullLogger()
+
+    async def create_collection_async(
+        self,
+        collection_name: str,
+        dimension_num: Optional[int] = None,
+    ) -> None:
+        """Creates a new collection.
+
+        Arguments:
+            collection_name {str} -- The name of the collection to create.
+            dimension_num {Optional[int]} -- The dimensionality of the embeddings. (default: {None})
+            Uses the default dimensionality when not provided
+
+        Returns:
+            None
+        """
+        if dimension_num is None:
+            dimension_num = self._default_dimensionality
+        if dimension_num > MAX_DIMENSIONALITY:
+            raise ValueError(
+                f"Dimensionality of {dimension_num} exceeds "
+                + f"the maximum allowed value of {MAX_DIMENSIONALITY}."
+            )
+        if dimension_num <= 0:
+            raise ValueError("Dimensionality must be a positive integer. ")
+
+        with self._connection_pool.connection() as conn:
+            conn.execute("CREATE EXTENSION IF NOT EXISTS vector")
```

Comment:
The C# implementation does not execute this statement anymore, because it only needs to be executed once after the database is created, and most importantly it seems that not all hosting can execute this statement, for example [Azure Cosmos DB for PostgreSQL](https://learn.microsoft.com/en-us/azure/cosmos-db/postgresql/howto-use-pgvector).

Reply:
Interesting, I didnt know that there were some hosting options with different statements for this. I can certainly remove it. Thanks for pointing this out! I have seen some of the work you have done/are doing against the C# side for the Postgres memory store. I'll have to take a look at what you currently have in the pipeline. Thanks @JadynWong